src/liblsquic/lsquic_enc_sess_ietf.c

```diff
@@ -553,8 +553,10 @@ gen_trans_params (struct enc_sess_iquic *enc_sess, unsigned char *buf,
             }
             cce->cce_seqno = seqno + 1;
             cce->cce_flags = CCE_SEQNO;
-            lsquic_generate_cid(&cce->cce_cid,
-                            enc_sess->esi_enpub->enp_settings.es_scid_len);
+
+            if (enc_sess->esi_enpub->enp_generate_cid) enc_sess->esi_enpub->enp_generate_cid(&cce->cce_cid);
```

Comment:
I would pass `es_scid_len` to the callback as I originally proposed.  The reason for this is that `es_scid_len` is used in the packet parsing code: the generated SCIDs *must be* that long.

Reply:
done. i put it as `unsigned`. any reason we shouldn't use uint_fast8_t given the max connection ID length is 20?

Reply:
It does not matter for the callback argument type.  We use integer types from `<stdint.h>` very sparingly in "lsquic.h".  Too, the type should match `es_scid_len`, which is `unsigned int`.

`lsquic_cid.len` is `uint_fast8_t`, which is what's important.